homeassistant/components/broadlink/remote.py

```diff
@@ -233,27 +155,17 @@ async def async_send_command(self, command, **kwargs):
             raise ValueError(err_msg)
 
         at_least_one_sent = False
-        for _, codes in product(range(repeat), code_list):
-            if at_least_one_sent:
-                await asyncio.sleep(delay)
-
-            if len(codes) > 1:
-                code = codes[self._flags[subdevice]]
-            else:
-                code = codes[0]
+        for _ in range(repeat):
+            for code in self._store.toggled_codes(code_list, subdevice):
 
-            try:
-                await device.async_request(device.api.send_data, code)
-            except (BroadlinkException, OSError) as err:
-                _LOGGER.error("Error during %s: %s", service, err)
-                break
+                if at_least_one_sent:
+                    await asyncio.sleep(delay)
 
-            if len(codes) > 1:
-                self._flags[subdevice] ^= 1
-            at_least_one_sent = True
-
-        if at_least_one_sent:
-            self._flag_storage.async_delay_save(self._get_flags, FLAG_SAVE_DELAY)
+                try:
+                    await device.async_request(device.api.send_data, code)
+                except (BroadlinkException, OSError) as err:
+                    _LOGGER.error("Error during %s: %s", service, err)
+                    break
```

Comment:
This breaks only the inner loop. We could use `product(range(repeat), self._store.toggled_codes(code_list, subdevice))` to make it break both of them.

Reply:
Actually we should just repeat before we turn the commands into codes instead.

Reply:
> Before [product()](https://docs.python.org/3/library/itertools.html#itertools.product) runs, it completely consumes the input iterables, keeping pools of values in memory to generate the products. Accordingly, it is only useful with finite inputs.

So it won't work since the codes will end up being toggled directly. Might make sense to just toggle manually.